test/integration/consul-container/test/multiport/explicit_destination_test.go

```diff
@@ -0,0 +1,273 @@
+	require.NoError(t, err)
+
+	n2 := cluster.Agents[2]
+	// Create a service and dataplane
+	clientDataplane, err := createServiceAndDataplane(t, n2, "static-client-workload", "static-client", 8080, 8079, []int{libcluster.ServiceUpstreamLocalBindPort})
+	require.NoError(t, err)
+
+	return serverDataplane, clientDataplane
+}
+
+func createServiceAndDataplane(t *testing.T, node libcluster.Agent, proxyID, serviceName string, httpPort, grpcPort int, serviceBindPorts []int) (*libcluster.ConsulDataplaneContainer, error) {
+	// Do some trickery to ensure that partial completion is correctly torn
+	// down, but successful execution is not.
+	var deferClean utils.ResettableDefer
+	defer deferClean.Execute()
+
+	// Create a service and proxy instance
+	svc, err := libservice.NewExampleService(context.Background(), serviceName, httpPort, grpcPort, node)
+	if err != nil {
+		return nil, err
+	}
+	deferClean.Add(func() {
+		_ = svc.Terminate()
+	})
+
+	// Create Consul Dataplane
+	dp, err := libcluster.NewConsulDataplane(context.Background(), proxyID, "0.0.0.0", 8502, serviceBindPorts, node)
+	require.NoError(t, err)
+	deferClean.Add(func() {
+		_ = dp.Terminate()
+	})
+
+	// disable cleanup functions now that we have an object with a Terminate() function
+	deferClean.Reset()
+
+	return dp, nil
+}
+
+func createServerServicesAndWorkloads(t *testing.T, resourceClient *rtest.Client, ipAddress string) *pbresource.Resource {
+	serverService := rtest.ResourceID(&pbresource.ID{
+		Name: "static-server-service",
+		Type: &pbresource.Type{
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Service",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).WithData(t, &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"static-server"}},
+		Ports: []*pbcatalog.ServicePort{
+			{TargetPort: "tcp", Protocol: pbcatalog.Protocol_PROTOCOL_TCP},
+			{TargetPort: "mesh", Protocol: pbcatalog.Protocol_PROTOCOL_MESH},
+		},
+	}).Write(t, resourceClient)
+
+	workloadPortMap := map[string]*pbcatalog.WorkloadPort{
+		"tcp": {
+			Port: 8080, Protocol: pbcatalog.Protocol_PROTOCOL_TCP,
+		},
+		"mesh": {
+			Port: 20000, Protocol: pbcatalog.Protocol_PROTOCOL_MESH,
+		},
+	}
+
+	rtest.ResourceID(&pbresource.ID{
+		Name: "static-server-workload",
+		Type: &pbresource.Type{
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Workload",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).
+		WithData(t, &pbcatalog.Workload{
+			Addresses: []*pbcatalog.WorkloadAddress{
+				{Host: ipAddress},
+			},
+			Ports:    workloadPortMap,
+			Identity: "static-server-identity",
+		}).
+		Write(t, resourceClient)
+	return serverService
+}
+
+func createClientResources(t *testing.T, resourceClient *rtest.Client, staticServerRef *pbresource.Resource, ipAddress string) {
```

Comment:
It might be clearer to rename `staticServerRef` since it isn't a reference.

Reply:
fixed in 4893908d568cc66fd254a9639657bc0e2a745bfd